### PR TITLE
(maint) update to clojure 1.9

### PR DESCRIPTION
**Pull request by jonathannewman**
This updates to clojure 1.9, and updates all the clojure space
dependencies to the latest.

In addition, it updates ring-mock to the new repo.

### CODE REVIEW

**Comment by pcarlisle** [comment]
so many things in here, i really have no way of knowing if this is okay

File: project.clj [hunk 1]
```diff
@@ -84,13 +84,13 @@
                          [trptcolin/versioneer "0.2.0"]
                          [io.dropwizard.metrics/metrics-core ~dropwizard-metrics-version]
                          [io.dropwizard.metrics/metrics-graphite ~dropwizard-metrics-version]
-                         [metrics-clojure "2.6.1"]
+                         [metrics-clojure "2.10.0"]
                          [org.ow2.asm/asm-all "5.0.3"]
-                         [honeysql "0.6.3"]
-                         [org.postgresql/postgresql "42.2.0"]
+                         [honeysql "0.9.2"]
```
Comment (pcarlisle) [inline project.clj]: this lists multiple breaking changes in its changelog and needs to be checked by the teams that use it (I know at least of puppetdb)

**Comment by MikaelSmith** [comment]
Call it a 2.0.0, and we'll figure it out from there?

File: project.clj (hunk 1 above)
Comment (jonathannewman) [reply project.clj]: I can revert it.  Thoughts?

Comment (jonathannewman) [reply project.clj]: I'll check it against the orchestrator too, which uses honeysql

**Comment by jonathannewman** [comment]
@pcarlisle I can pair it down to just the clojure namespace items if that would make it easier.  I updated the tag as 2.0.0-SNAPSHOT.

**Comment by jonathannewman** [comment]
@pcarlisle @MikaelSmith  I backed out some of the changes after running into issues integrating it into projects.  I think this should be a reasonable set to work from .  Thoughts?

**Comment by MikaelSmith** [comment]
I don't see anything concerning. I'm ok with a little work in other projects to pick up new versions of dependencies.

**Comment by ALawley** [comment]
👍 Image

**Pull request merged**